### PR TITLE
Fix #24818 by updating linalg_ops.py

### LINKED ISSUE #24818: eigvalsh documentation error
<em>Please make sure that this is a documentation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:doc_template</em>

**System information**
- TensorFlow version: 1.12
- Doc Link:

https://www.tensorflow.org/api_docs/python/tf/linalg/eigvalsh

**Describe the documentation issue**

Note: If your program backpropagates through this function, you should replace it with a call to tf.linalg.eigvalsh (possibly ignoring the second output) to avoid computing the eigen decomposition twice. This is because the eigenvectors are used to compute the gradient w.r.t. the eigenvalues. See _SelfAdjointEigV2Grad in linalg_grad.py.

This warning is quite ambiguous since
1) ling.eigvalsh returns only one output, while linalg.eigh returns two outputs
2) it circularly references itself

**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

Yes if it devs agree that there's an issue.

### PR DESCRIPTION
**Pull request by bharatr21**
Fix #24818 by replacing the circular reference to `tf.linalg.eigvalsh`  with `tf.linalg.eigh`  [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/linalg_ops.py#L338) which also returns two outputs.
I believe that `tf.linalg.eigh` is the correct equivalent to resolve:
1. Circular Reference
2. Two outputs instead of one output, highlighted as "(possibly ignoring the second output)" in [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/linalg_ops.py#L338)

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @annarev: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Review by EmpRamses — APPROVED** [review]
I have got an InvalidArgumentError when I follow the notice in api page. This change solve my problem.

InvalidArgumentError (see above for traceback): Got info = 31 for batch index 0, expected info = 0. Debug_info = heevd
	 [[node SelfAdjointEigV2 (defined at /data/trojai/trojan_detection/src/solver.py:34)  = SelfAdjointEigV2[T=DT_FLOAT, compute_v=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](MatMul)]]

**Review by penpornk — APPROVED** [review]
Thank you for the fix! I believe you are correct. We had a renaming mistake from https://github.com/tensorflow/tensorflow/commit/f1fd53748b99532b2572b8909efcd4f5c06ce28d. It used to be `tf.self_adjoint_eig` which should have been renamed to `tf.linalg.eigh` instead of `tf.linalg.eigvalsh`.
